azurerm/resource_arm_netapp_volume.go

```diff
@@ -0,0 +1,357 @@
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmNetAppVolumeRead(d, meta)
+}
+
+func resourceArmNetAppVolumeRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Netapp.VolumeClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	accountName := id.Path["netAppAccounts"]
+	poolName := id.Path["capacityPools"]
+	name := id.Path["volumes"]
+
+	resp, err := client.Get(ctx, resourceGroup, accountName, poolName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] NetApp Volumes %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading NetApp Volumes %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("account_name", accountName)
+	d.Set("pool_name", poolName)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if props := resp.VolumeProperties; props != nil {
+		d.Set("creation_token", props.CreationToken)
+		d.Set("service_level", props.ServiceLevel)
+		d.Set("subnet_id", props.SubnetID)
+
+		if props.UsageThreshold != nil {
+			d.Set("usage_threshold", *props.UsageThreshold/1073741824)
+		}
+		if props.ExportPolicy != nil {
+			if err := d.Set("export_policy_rule", flattenArmExportPolicyRule(props.ExportPolicy.Rules)); err != nil {
+				return fmt.Errorf("Error setting `export_policy_rule`: %+v", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func resourceArmNetAppVolumeDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Netapp.VolumeClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	accountName := id.Path["netAppAccounts"]
+	poolName := id.Path["capacityPools"]
+	name := id.Path["volumes"]
+
+	_, err = client.Delete(ctx, resourceGroup, accountName, poolName, name)
+	if err != nil {
+		return fmt.Errorf("Error deleting NetApp Volume %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	return waitForNetAppVolumeToBeDeleted(ctx, client, resourceGroup, accountName, poolName, name)
+}
+
+func waitForNetAppVolumeToBeDeleted(ctx context.Context, client *netapp.VolumesClient, resourceGroup, accountName, poolName, name string) error {
+	log.Printf("[DEBUG] Waiting for NetApp Volume Provisioning Service %q (Resource Group %q) to be deleted", name, resourceGroup)
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"200", "202"},
+		Target:  []string{"404"},
+		Refresh: netappVolumeDeleteStateRefreshFunc(ctx, client, resourceGroup, accountName, poolName, name),
+		Timeout: 20 * time.Minute,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
```

Comment:
can we not use the Future to handle this?

Reply:
Probably not. Because there is an issue here https://github.com/Azure/azure-sdk-for-go/issues/6374

Reply:
That is for the pool, have you checked if volume exhibits the same behaviour?

Reply:
Yes. I already double checked before against this issue. both volume and pool have the same issue. So I also mentioned the volume also exists the same issue in this issue link https://github.com/Azure/azure-sdk-for-go/issues/6374

![image](https://user-images.githubusercontent.com/19754191/69391275-41ecdf00-0d0d-11ea-95f0-1a1f7e992456.png)

Reply:
Could we link back to that issue from here/create a new one update that to reflect it also affects this call?